esp/bindings/SOAP/client/soapclient.hpp

```diff
@@ -45,6 +49,20 @@ class esp_http_decl CSoapClient : implements ISoapClient, public CInterface
     CSoapClient(){m_disableKeepAlive=false;}
     CSoapClient(ITransportClient* transportclient){m_transportclient.setown(transportclient); m_disableKeepAlive=false;}
     virtual ~CSoapClient() {};
+
+    void setConnectTimeoutMs(unsigned val)
+    {
+        connectTimeoutMs = val;
+        if (m_transportclient)
+            static_cast<IHttpClient*>(m_transportclient.get())->setConnectTimeOutMs(connectTimeoutMs); //until we support something other than soap over http, static_cast
```

Comment:
Transport client could be anything... for instance it could post to a transaction queue.  Connection and read timeouts may not apply to all.